src/VisualStudio/Core/Impl/CodeModel/RootCodeModel.cs

```diff
@@ -81,7 +81,7 @@ private Compilation GetCompilation()
                 }
 
                 var hostProject = ((VisualStudioWorkspaceImpl)Workspace).DeferredState.ProjectTracker.GetProject(_projectId);
-                var projectCodeModel = (ProjectCodeModel)((IProjectCodeModelProvider)hostProject).ProjectCodeModel;
+                var projectCodeModel = (ProjectCodeModel)hostProject.ProjectCodeModel;
```

Comment:
i really don't like all tehse casts for this sort of hting.

Reply:
Yeah, I don't either, see commentary on IProjectCodeModelProvider.cs above.